dspace-api/src/main/java/org/dspace/util/SolrUpgradeStatistics6.java

```diff
@@ -0,0 +1,662 @@
+         * Otherwise, the following sequence will be applied in order to optimize hibernate caching.
+         * 
+         * Communities and Collections - retain in the cache since each is likely to be re-used
+         * Items - retain in the cache until a new item is processed
+         * Bitstreams - retain in the cache until a new bitstream is processed
+         */
+        private void run() throws SolrServerException, SQLException, IOException {
+                runReport();
+                logTime(false);
+                for(int processed = updateRecords(MIGQUERY); (processed != 0) && (numProcessed < numRec);  processed = updateRecords(MIGQUERY)){
+                        printTime(numProcessed, false);
+                        batchUpdateStats();
+                        if (context.getCacheSize() > CACHE_LIMIT) {
+                                refreshContext();
+                        }
+                }                
+                printTime(numProcessed, true);
+                
+                if (numProcessed > 0) {
+                        runReport();
+                }
+        }
+
+        
+        
+        /*
+         * Update records associated with a particular object id
+         * @param query Query to retrieve all of the statistics records associated with a particular object
+         * @param field Field to use for grouping records
+         * @return number of items processed.  0 indicates that no more work is available (or the max processed has been reached).
+         */
+        private int updateRecords(String query) throws SolrServerException, SQLException, IOException {
+                int initNumProcessed = numProcessed;
+                SolrQuery sQ = new SolrQuery();
+                sQ.setQuery(query);
+                sQ.setRows(batchSize);
+                
+                //Ensure that items are grouped by id
+                //Sort by id fails due to presense of id and string fields. The ord function seems to help
+                sQ.addSort("type", SolrQuery.ORDER.desc);
+                sQ.addSort("scopeType", SolrQuery.ORDER.desc);
+                sQ.addSort("ord(owningItem)", SolrQuery.ORDER.desc);
+                sQ.addSort("id", SolrQuery.ORDER.asc);
+                sQ.addSort("scopeId", SolrQuery.ORDER.asc);
+                
+                QueryResponse sr = server.query(sQ);
+                SolrDocumentList sdl = sr.getResults();
+                
+                for(int i=0; i<sdl.size() && (numProcessed < numRec); i++) {
+                        SolrDocument sd = sdl.get(i);
+                        SolrInputDocument input = ClientUtils.toSolrInputDocument(sd);
+                        input.remove("_version_");
+                        for(FIELD col: FIELD.values()) {
+                                mapField(input, col);
+                        }
+                        input.addField("dspaceMig", "6.1");
```

Comment:
This should likely be updated to "6.x" (if that's a valid value). The migration here is not specific to 6.1 in any way, and obviously 6.1 was already released.

By the way, does this "dspaceMig" field need to be defined in our Solr Schema?  I notice it is not defined.

Reply:
Apparently we *do* need to define field 'dspaceMig'.  

1. I built an instance of this PR
1. installed it
1. moved the empty statistics core
1. unpacked an archived snapshot of a production 5.2 statistics core
1. copied the content of the config directory from the saved core
1. tried to run 'bin/dspace solr-upgradeD6-statistics' in it

and immediately got this:

No index name provided, defaulting to : statistics
Connecting to http://mhw.ulib.iupui.edu:8080/PR1810/solr/statistics

=================================================================
         *** Statistics Records with Legacy Id ***

Exception: undefined field dspaceMig
org.apache.solr.client.solrj.impl.HttpSolrServer$RemoteSolrException: undefined field dspaceMig

Reply:
I added <field name='dspaceMig' type='string'.../> and now migration is proceeding.